CMake bug alpaka_add_library(${_TARGET} STATIC) with Clang as CUDA compiler

If I use `alpaka_add_library(${TARGET_NAME} STATIC  file.cpp)` and compile my project with Clang as CUDA compiler, I get following error:

```bash
-- Configuring done
CMake Error at alpaka/cmake/addLibrary.cmake:95 (ADD_LIBRARY):
  Cannot find source file:
    STATIC
  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .h .hh .h++
  .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .ispc
Call Stack (most recent call first):
  cmake/cuplaTargetHelper.cmake:27 (alpaka_add_library)
  CMakeLists.txt:101 (createCuplaTarget)
CMake Error at alpaka/cmake/addLibrary.cmake:95 (ADD_LIBRARY):
  No SOURCES given to target: cupla
Call Stack (most recent call first):
  cmake/cuplaTargetHelper.cmake:27 (alpaka_add_library)
  CMakeLists.txt:101 (createCuplaTarget)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
Full log: https://gitlab.com/hzdr/crp/cupla/-/jobs/1422713016

The problem is the following lines of code: https://github.com/alpaka-group/alpaka/blob/394eb4e5eea81b1a418bd33f893f53c97c1e77bb/cmake/addLibrary.cmake#L95-L101

`libaryType` should be on the second position. I expected, that this bug only affected alpaka `0.6.1` and older, because we replaced the affected code in alpaka `0.7.0`.

--- a/cmake/addLibrary.cmake
+++ b/cmake/addLibrary.cmake
@@ -1,5 +1,5 @@
 #
-# Copyright 2015-2019 Benjamin Worpitz, Maximilian Knespel
+# Copyright 2015-2021 Benjamin Worpitz, Maximilian Knespel, Simeon Ehrig
 #
 # This file is part of alpaka.
 #
@@ -94,10 +94,10 @@ MACRO(ALPAKA_ADD_LIBRARY libraryName)
             ENDFOREACH()
             ADD_LIBRARY(
                 ${libraryName}
-                ${sourceFileNames}
                 ${libraryType}
                 ${excludeFromAll}
                 ${optionArguments}
+                ${sourceFileNames}
             )
         ELSE()
             FOREACH( _file ${ARGN} )

